central/pruning/pruning.go

```diff
@@ -485,6 +494,127 @@ func (g *garbageCollectorImpl) collectImages(config *storage.PrivateConfig) {
 	}
 }
 
+func (g *garbageCollectorImpl) collectClusters(config *storage.PrivateConfig) {
+	if !features.DecommissionedClusterRetention.Enabled() {
+		return
+	}
+
+	if lastClusterPruneTime.Add(clusterGCFreq).After(time.Now()) {
+		// Only cluster pruning if it's been at least clusterGCFreq since last run
+		return
+	}
+	defer func() {
+		lastClusterPruneTime = time.Now()
+	}()
+
+	clusterRetention := config.GetDecommissionedClusterRetention()
+	retentionDays := int64(clusterRetention.GetRetentionDurationDays())
+	if retentionDays == 0 {
+		log.Info("[Cluster Pruning] pruning is disabled.")
+		return
+	}
+
+	lastUpdateTime, err := types.TimestampFromProto(clusterRetention.GetLastUpdated())
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	if timeutil.TimeDiffDays(time.Now(), lastUpdateTime) < 1 {
+		// Allow 24 hr grace period after a change in config
+		log.Info("[Cluster Pruning] skipping pruning to allow 24 hr grace period after a recent change in cluster retention config.")
+		return
+	}
+
+	configCreationTime, err := types.TimestampFromProto(clusterRetention.GetCreatedAt())
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	if timeutil.TimeDiffDays(time.Now(), configCreationTime) < int(retentionDays) {
+		// In this case, the deployments that became unhealthy after config creation would also be unhealthy for fewer than retention days
+		// and pruning can be skipped
+		log.Info("[Cluster Pruning] skipping pruning as retention period only starts after upgrade to version 3.71.0.")
+		return
+	}
+
+	allClusterCount, err := g.clusters.CountClusters(pruningCtx)
+	if err != nil {
+		log.Errorf("[Cluster Pruning] error counting clusters: %v", err)
+	}
+
```

Comment:
check `allClusterCount == 0`

Reply:
No, you don't need to have a secured cluster running in the cluster where central is. (And this will always be the case in a managed services setting, for example.)

Reply:
Ah gotcha. I added the check in.